app/src/main/java/com/example/pix/home/fragments/ChatsFragment.java

```diff
@@ -76,10 +83,64 @@ public void onLoadMore(int page, int totalItemsCount, RecyclerView view) {
             };
 
             rvChats.addOnScrollListener(scroll);
+
+            // Handle what happens when a Chat is swiped
+            ItemTouchHelper.SimpleCallback simpleCallback = new ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.RIGHT) {
+                @Override
+                public boolean onMove(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder, @NonNull RecyclerView.ViewHolder target) {
+                    return false;
+                }
+
+                @Override
+                public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction) {
+                    final int position = viewHolder.getAdapterPosition();
+                    // Get the Chat that was swiped
+                    Chat toDelete = chats.get(position);
+                    try {
+                        // Delete any messages from this Chat
+                        deleteMessages(toDelete);
+                        // Delete the Chat
+                        toDelete.delete();
+                    } catch (ParseException e) {
+                        e.printStackTrace();
+                        Toast.makeText(getContext(), "Error deleting Messages", Toast.LENGTH_SHORT).show();
+                        return;
+                    }
+                    // Update our RecyclerView and Adapter
+                    chats.remove(position);
+                    chatsAdapter.notifyItemRemoved(position);
+                }
+            };
+            ItemTouchHelper itemTouchHelper = new ItemTouchHelper(simpleCallback);
+            itemTouchHelper.attachToRecyclerView(rvChats);
         } catch (ParseException e) {
             Log.e("Error", "Error getting List of Chats", e);
             Toast.makeText(getContext(), "Error retrieving chats", Toast.LENGTH_SHORT).show();
         }
+    }
+
+    // Get a List of Messages from this Chat then delete each one of them
+    public void deleteMessages(Chat chat) throws ParseException {
```

Comment:
Question: does this mean the messages are deleted from the database? If I have a chat with you, and I deleted our chat on my end, does this mean you will also lose the chat in your app? If that's the correct understanding, you probably don't want to do that. You should just hide the chat on my end, without deleting the messages.

Reply:
I made it so that you "Archive" a Chat instead of deleting it. The User can then search for this Archived friend and re-open their old messages that way. I also set up a commit to the PopupHelper branch updating SearchAdapter, so that if we find an archived Chat, we Unarchive it before opening it, so that it becomes visible.